### PR TITLE
JetBrains: Improve search result UI/UX

### LINKED ISSUE #36703: JetBrains: Search results styling
All six types

### PR DESCRIPTION
**Pull request by philipp-spiess**
Closes #36703

I’m sorry that all of the changes are inside one commit but these were very intertwined (and I, again, haven't thought about this until after I have touched too many files 😢 )

This basically applies the Figma designs for all search result types to our component structure. I'm explaining all the changes in more detail in the Loom.

It also fixes an issue where calling event stream polyfill in a browser (like we do here inside Chromium) breaks our tooltip library. The reason for that is that it also polyfills `window.Event`, even though browsers support that already and it is not necessary to do for only applying headers properly. I've slightly changed the polyfill code to use native `Event` and `MessageEvent` which seems to work fine for me! (cc @tjdevries)

In the video I mention two issues which I checked against `main` to not be regressions. Created follow-up tasks for that:
- #36812
- #36813

## Test plan

Manually tested:

https://www.loom.com/share/6ba6f9a04f634ccc8f763b41bf36208a

## App preview:

- [Web](https://sg-web-ps-jetbrains-theming-result-list.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-wnxciiwbat.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

File: client/shared/src/polyfills/vendor/eventSource.js [hunk 1]
```diff
@@ -16,8 +16,15 @@ module.exports = function polyfillEventSource(headers) {
   fixedHeaders = { ...headers }
 
   global.EventSource = EventSource
-  global.MessageEvent = MessageEvent
-  global.Event = Event
+
+  // It's safe to use a browser implementation of Event and MessageEvent if we only polyfill to add
+  // support for additional header fields.
```
Comment (tjkandala) [inline client/shared/src/polyfills/vendor/eventSource.js:21]: Makes sense!

**Review by tjkandala — APPROVED** [review]
Looks great and `<SelectableSearchResult>` + `<SearchResultLayout>` is cool 😎

File: client/shared/src/polyfills/vendor/eventSource.js (hunk 1 above)
Comment (vdavid) [reply client/shared/src/polyfills/vendor/eventSource.js:21]: Wow, nice catch!

**Review by vdavid — APPROVED** [review]
Great job on refactoring the search results and structuring the CSS! And it looks really nice!

One additional thing that's very quick to do and would give a huge boost to the native feel is **setting the cursor to arrow** for the whole result list! With that in place, I think I'd believe that I'm seeing a native component. We could include it in this PR or do this change later, whichever you prefer.

**Comment by philipp-spiess** [comment]
> One additional thing that's very quick to do and would give a huge boost to the native feel is setting the cursor to arrow for the whole result list!

Oh yeah thanks for catching this regression!

**Comment by vdavid** [comment]
It's not a regression, actually, there's the same pointer icon on `main`.

**Comment by philipp-spiess** [comment]
> It's not a regression, actually, there's the same pointer icon on main.

Gotcha! I thought I've added this before haha my bad! Thanks for spotting regardless though 🙂

**Review by jjinnii — APPROVED** [review]
Looks really great!! In addition to the padding btw results + starred count that I mentioned in slack yesterday, I'm going to play around with the UI Inspector some more to see if we can get spacing + sizing to look closer to Find in Files. I will take some time to do that and create a new issue if I think there are any updates to be made.

**Comment by philipp-spiess** [comment]
@jjinnii Fixed the paddings between results and starred count:

Screenshot 2022-06-10 at 13 00 04

**Pull request merged**